docs/books/valhalla-mjolnir/src/01_valhalla_tiled_structure.md

```diff
@@ -0,0 +1,147 @@
+
+## Tile Size and Resolution
+
+The size of tiles varies by level:
+
+- Level 0 tiles are the largest, covering approximately 4 degrees × 4 degrees
+- Level 1 tiles cover 1 degree × 1 degree
+- Level 2 tiles are the smallest, covering 0.25 degrees × 0.25 degrees
+
+This variable resolution allows for efficient storage and retrieval of road network data at different scales. The world is divided into a grid based on these tile sizes, with rows and columns starting from the bottom left (-180, -90) and increasing to the top right (180, 90).
+
+## Code Example: Tile Coordinates Calculation
+
+Here's how Valhalla converts geographic coordinates to tile coordinates:
+
+```cpp
+// From midgard/tiles.h
+class Tiles {
+public:
+  /**
+   * Get the tile ID given the latitude and longitude.
+   * @param lat Latitude value.
+   * @param lng Longitude value.
+   * @return Returns the tile ID.
+   */
+  int32_t TileId(const float lat, const float lng) const;
+  
+  // ...
+};
+```
+
+The implementation uses a projection system to map geographic coordinates to tile IDs:
+
+```cpp
+// From midgard/tiles.cc
+int32_t Tiles::TileId(const float lat, const float lng) const {
+  // Get the indexes within the tiles and return the tile ID
+  const int32_t x = static_cast<int32_t>(floor((lng - tilebounds_.minx()) * inv_tile_size_));
+  const int32_t y = static_cast<int32_t>(floor((lat - tilebounds_.miny()) * inv_tile_size_));
+  return (y * ncolumns_) + x;
+}
+```
+
+## Tile Storage Format
+
+Valhalla stores tiles as binary files on disk. Each tile contains:
+
+1. **Header**: Contains metadata about the tile
+2. **Nodes**: Represent intersections in the road network
+3. **Directed Edges**: Represent road segments with direction-specific attributes
+4. **Edge Info**: Contains shared data for edges (e.g., shape, names)
+5. **Additional Data**: Restrictions, traffic signs, admin information, etc.
+
+The binary format is designed for efficient storage and quick loading. Here's the structure of a tile header:
+
+```cpp
+// From baldr/graphtileheader.h
+class GraphTileHeader {
+public:
+  // ...
+  
+  uint32_t graphid_;                 // Tile ID within the hierarchy
+  uint32_t date_created_;            // Date created
+  uint32_t node_count_;              // Number of nodes
+  uint32_t directed_edge_count_;     // Number of directed edges
+  // ... many more fields ...
+};
+```
+
+The tiles are stored on disk with a directory structure that reflects their hierarchy level and ID. For example, a tile with ID 756425 at level 2 would be stored at `/2/000/756/425.gph`.
+
+## Tile Connectivity
+
+Tiles connect to adjacent tiles through "transition" edges and nodes. These special connections allow routes to seamlessly cross tile boundaries. Additionally, connections between hierarchy levels are maintained through "transition up/down" edges, which allow routes to move between local, regional, and highway networks.
```

Comment:
this is not correct. adjacent tiles in the same hierarchy level connect across tile boundaries simply via node ids. that is, an edge has and end node id and that id contains the tile identifier within which its end node will be found. transition edges are used to transition at a node from one level of the hierarchy to its corresponding node on another hierarchy level (same node in physical space but duplicated across the hierarchy for the sake of topology).